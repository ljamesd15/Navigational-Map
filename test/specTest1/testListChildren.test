# testListChildren.test file
# This tests whether Graph.java correctly stores the connections of the
# 	characters.

# Create three graphs
CreateGraph testGraph1
CreateGraph testGraph2
CreateGraph testGraph3

# Add some edges testing one character doesn't get added twice.
AddEdge testGraph1 Spider-Man Hulk BookA
AddEdge testGraph1 Spider-Man Thor BookA

# Add some edges testing one character doesn't get added twice.
AddEdge testGraph2 Daredevil Deadpool BookB
AddEdge testGraph2 Daredevil Deadpool BookC

# Add some edges multiple connections from different characters.
AddEdge testGraph3 Magneto Hulk BookD
AddEdge testGraph3 Thor Spider-Man BookD
AddEdge testGraph3 Daredevil Thor BookD
AddEdge testGraph3 Hulk Deadpool BookD
AddEdge testGraph3 Thor Spider-Man BookD
AddEdge testGraph3 Deadpool Thor BookD

# Print out the children nodes from each graph
ListChildren testGraph1 Spider-Man
ListChildren testGraph2 Daredevil
ListChildren testGraph2 Deadpool
ListChildren testGraph3 Thor
ListChildren testGraph3 Hulk

# Attempt to get children from a graph which has not been created.
ListChildren graphDoesNotExist Hulk