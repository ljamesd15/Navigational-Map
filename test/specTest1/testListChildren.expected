# testListChildren.test file
# This tests whether Graph.java correctly stores the connections of the
# 	characters.

# Create three graphs
created graph testGraph1
created graph testGraph2
created graph testGraph3

# Add some edges testing one character doesn't get added twice.
added edge BookA from Spider-Man to Hulk in testGraph1
added edge BookA from Spider-Man to Thor in testGraph1

# Add some edges testing one character doesn't get added twice.
added edge BookB from Daredevil to Deadpool in testGraph2
added edge BookC from Daredevil to Deadpool in testGraph2

# Add some edges multiple connections from different characters.
added edge BookD from Magneto to Hulk in testGraph3
added edge BookD from Thor to Spider-Man in testGraph3
added edge BookD from Daredevil to Thor in testGraph3
added edge BookD from Hulk to Deadpool in testGraph3
The edge BookD from Thor to Spider-Man was unsuccessfully added to testGraph3
added edge BookD from Deadpool to Thor in testGraph3

# Print out the children nodes from each graph
the children of Spider-Man in testGraph1 are: Hulk(BookA) Thor(BookA)
the children of Daredevil in testGraph2 are: Deadpool(BookB) Deadpool(BookC)
the children of Deadpool in testGraph2 are:
the children of Thor in testGraph3 are: Spider-Man(BookD)
the children of Hulk in testGraph3 are: Deadpool(BookD)

# Attempt to get children from a graph which has not been created.
The graph 'graphDoesNotExist' does not exist.